include/FanFicArchive.py

```diff
@@ -0,0 +1,85 @@
+    tag_list = {}
+    for i in tag_line:
+        if ":" in i: bracket, item = i.split(": ")[0:2]; tag_list[bracket] = item.rstrip(' ')
+        else: tag_list[['Language', 'Genre', 'Characters'][tag_line.index(i)-1]] = i
+    tag_list['Characters'] = tag_list['Characters'].split(', ')
+    tag_list['Genre'] = tag_list['Genre'].split('/')
+    del tag_list['id']
+    author = text.find("div", id="profile_top").find("a", class_="xcontrast_txt", title=False, target=False) # tag filtering
+    userid, username = author.get("href").split("/")[-2:] # the text in the tag, unlike the last line
+    summary = text.find("div", class_="xcontrast_txt", style=True).get_text() # tag filtering
+    chapter_list = {}
+    for i in range(1, int(tag_list['Chapters'])+1): # loop through each chapter
+        r = request.urlopen("https://www.fanfiction.net/s/"+ff_id+"/"+str(i)).read()
+        storyid = match(".*storytextid=([0-9]+)", str(r)).group(1) # can't get it any other way
+        text = BeautifulSoup(r, "lxml")
+        story = text.find("div", class_="storytext xcontrast_txt nocopy", id="storytext") # literal lines
+        chapter_list[i] = [storyid, story]
+    return ff_id, userid, username, summary, tag_list, chapter_list
+
+def da_rip(user_id, item_id): # name of literature can replace user_id, remember to bypass mature filters
+    fanfic_check(user_id, da_rip); fanfic_check(item_id, da_rip)
+    text = fanfic_req("https://deviantart.com/art/"+user_id+"-"+itemid) # their url system is shit
+    title = text.find("title").get_text() # title of the text
+    pretext = text.find("li", class_="author") # description bar
+    artist = pretext.find("a", class_="u regular username").get_text()
+    upload_time = pretext.find("span", title=True).get_text()
+    summary = text.find("meta", property="og:description").get("content") # metadata at the top
+    idnum = {}
+    for i in ["itemid", "splitid", "ownerid"]: # comments section
+        idnum[i] = text.find("input", {"type":"hidden", "name":i}).get("value")
+    type_tags = text.find("meta", {"name":"keywords"}).get("content").split(", ") # metadata at the top
+    art_tags = [i.get_text()for i in text.findall("a", class_="discoverytag")] # description ber
+    img = text.find("meta", property="og:image").get("content") # metadata at the top
+    dick = text.find("div", class_="grf-indent").find("div", class_="text").contents[:-2] # js in text is not cool
+    return title, artist, summary, idnum, type_tags, art_tags, img, dick
+
+def watt_rip(watt_id): # 100% queries, WTF? v4 is proprietary, WTF!
+    fanfic_check(watt_id, "watt_rip")
+    book = requests.get('https://www.wattpad.com/apiv2/info?id=' + watt_id, headers={'User-Agent': 'Mozilla/5.0'}).json()
+    chapter_list = book['group']; result = {"tags": book["tags"].split(" ")}
+    for i in ["cover", "description", "groupId", "author", "completed"]: result[i] = book[i]
```

Comment:
Might have been easier to do `result.update(book)`?

Reply:
dunno what you mean